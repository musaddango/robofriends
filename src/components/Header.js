import React from 'react';
import CounterButton from './CounterButton';

export default class Header extends React.Component {

    shouldComponentUpdate(prev, next){
        return false;
    }
    render(){
        console.log('Header')
        return (<>
                    <h1 className='f1'>RoboFriends</h1>
                    <CounterButton color={'red'}/>
                </>
                )
    }
}