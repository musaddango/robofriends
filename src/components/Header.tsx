import React from 'react';
import CounterButton from './CounterButton.tsx';

export default class Header extends React.Component {

    // shouldComponentUpdate(prev, next){
    //     return false;
    // }
    render(): React.JSX.Element{
        return (<>
                    <h1 className='f1'>RoboFriends</h1>
                    <CounterButton color={'red'}/>
                </>
                )
    }
}