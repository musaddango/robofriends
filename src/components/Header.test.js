import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from './Header';

it('expect to render Card component', ()=>{
    const wrapper = shallow(<Header />)

    expect(toJson(wrapper)).toMatchSnapshot();
});