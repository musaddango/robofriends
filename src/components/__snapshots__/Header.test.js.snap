// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component 1`] = `
<Fragment>
  <h1
    className="f1"
  >
    RoboFriends
  </h1>
  <CounterButton
    color="red"
  />
</Fragment>
`;
